sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretClientBuilder.java

```diff
@@ -177,8 +179,14 @@ public SecretClient buildClient() {
      * and {@link #retryPolicy(RetryPolicy)} have been set.
      */
     public SecretAsyncClient buildAsyncClient() {
+        return new SecretAsyncClient(buildInnerClient());
+    }
+
+
+    private SecretClientImpl buildInnerClient() {
         Configuration buildConfiguration =
             (configuration == null) ? Configuration.getGlobalConfiguration().clone() : configuration;
+
         URL buildEndpoint = getBuildEndpoint(buildConfiguration);
```

Comment:
Should we follow the same thing as KV keys here and return a String instead of an URL.
        String buildEndpoint = getBuildEndpoint(buildConfiguration);